% NEW based on COMT-MB data



\documentclass[11pt]{article}
\usepackage{graphicx, subfig}
\usepackage{float}
\pagenumbering{arabic}
\usepackage{enumerate}
\usepackage{Sweave}
\usepackage{booktabs}
\usepackage[table]{xcolor}
\usepackage{framed}
\usepackage{longtable}
\usepackage{tablefootnote}
\usepackage{amsmath}
\usepackage{hyperref}


\begin{document}
\input{comt_correlations_ontology_general_v3atemp-concordance}
 

This program has the unique ID: 405cc2c5dd4bd8e0983992907dcb72db.

\hspace{-1.5em}Gene Network\\
COMT expression correlations in Four Bartin Areas\\
Fall, 2017\\






{\textit {COMT}} catalyzes degradation of catecholamines including dopamine, norepinephrine and epinephrine.\\ 
%latex.default(url.Tab, file = "", caption = c("Gene Coexpression Databases"),     caption.loc = c("top"), colname = c("Name", "URL"), rowlabel = c(""),     colnamesTexCmd = "bfseries", booktabs = TRUE, label = c("tabcoexpressurls"),     na.blank = TRUE, vbar = FALSE, longtable = TRUE, table.env = TRUE,     center = c("center"), continued = c("Gene Coexpression Databases Continued"),     first.hline.double = TRUE, append = FALSE)%
\setlongtables\begin{longtable}{lll}\caption{Gene Coexpression Databases} \tabularnewline
\toprule
\multicolumn{1}{l}{\bfseries }&\multicolumn{1}{c}{\bfseries site}&\multicolumn{1}{c}{\bfseries addr}\tabularnewline
\midrule
\endfirsthead\caption[]{\em (Gene Coexpression Databases Continued)} \tabularnewline
\midrule
\multicolumn{1}{l}{\bfseries }&\multicolumn{1}{c}{\bfseries site}&\multicolumn{1}{c}{\bfseries addr}\tabularnewline
\midrule
\endhead
\midrule
\endfoot
\label{tabcoexpressurls}
1&COXPRESdb&http://coxpresdb.jp/\tabularnewline
2&OMICtools&https://omictools.com/\tabularnewline
3&Coexpedia&http://www.coexpedia.org/search.php\tabularnewline
4&GeneFriends&http://www.genefriends.org/RNAseq/\tabularnewline
5&Illumina Probes&http://www.genomequebec.mcgill.ca/compgen/integratedvervetgenomics/transcriptome/Illumina/allprobes.html\tabularnewline
6&Gibbs Expression Data&https://www.ncbi.nlm.nih.gov/geo/query\tabularnewline
7&Train Online&https://www.ebi.ac.uk/training/online/course/arrayexpressdiscoverfunctionalgenomicsdataqui/references\tabularnewline
\bottomrule
\end{longtable}











%%%%%%%%%%%%%%% Statistics



c(2055, 2047, 2043, 2027)\\





[1] 2055[1] 1969[1] 1[1] 427


[1] 2113[1] 1999[1] 1651[1] 462


[1] 2047[1] 1926[1] 1[1] 432


[1] 2101[1] 1950[1] 1635[1] 466
%%%%%%%%%%%%%% for tempor and pons below use the code for recovering Chr ans start positions
%%%%%%%%%%%%%% immeditately above for cbell



[1] 2043[1] 1947[1] 1[1] 426

[1] 2098[1] 1975[1] 1638[1] 460

[1] 2027[1] 1934[1] 1[1] 419

[1] 2084[1] 1963[1] 1630[1] 454

c(20000, 19573, 19538)

[1] 359[1] 321
There are 109 entries from among the original 20,000 that do not have valid Chromosome names, and 142 that do not have valid start positions.

[1] 363[1] 327
There are 108 entries from among the original 20,000 that do not have valid Chromosome names, and 140 that do not have valid start positions.


[1] 356[1] 319
There are 109 entries from among the original 20,000 that do not have valid Chromosome names, and 142 that do not have valid start positions.

[1] 357[1] 312
There are 109 entries from among the original 20,000 that do not have valid Chromosome names, and 143 that do not have valid start positions.




The analyses of effects by Chromosome and by Starting Position are based on the counts as shown in the table below.\\

c(427, 375, 906, 359, 321, 318, 109, 19891, 19858)


\end{document}


 
