% Use this file as an \include to put it at the top of Sweave/R generated latex tables.

\documentclass[letterpaper,12pt]{article}

%\usepackage{dgjournal}


%%% Note: remove the next two lines to submit article %%%%%%%%%%
\usepackage{setspace}
%\linespread{2}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%       

%% The mathptmx package is recommended for Times compatible math symbols.
%% Use mtpro2 or mathtime instead of mathptmx if you have the commercially
%% available MathTime fonts.
%% Other options are txfonts (free) or belleek (free) or TM-Math (commercial)
\usepackage{mathptmx}
\usepackage{tensor}
\usepackage{nicefrac}
\usepackage{units}
\usepackage{mathtools}
\usepackage{mathrsfs}
\DeclareMathAlphabet{\mathpzc}{OT1}{pzc}{m}{it}
\usepackage{nth}
\usepackage{epsfig}
\usepackage[makeroom]{cancel}
\newcommand\Ccancel[2][black]{\renewcommand\CancelColor{\color{#1}}\cancel{#2}}
\newcommand\Bcancel[2][black]{\renewcommand\CancelColor{\color{#1}}\bcancel{#2}}

\usepackage{geometry}
\usepackage{changepage}



\usepackage[T1]{fontenc}
\usepackage{makecell}
\newcolumntype{x}[1]{>{\centering\arraybackslash}p{#1}}

\usepackage{tikz}
\newcommand\diag[4]{%
  \multicolumn{1}{p{#2}|}{\hskip-\tabcolsep
  $\vcenter{\begin{tikzpicture}[baseline=0,anchor=south west,inner sep=#1]
  \path[use as bounding box] (0,0) rectangle (#2+2\tabcolsep,\baselineskip);
  \node[minimum width={#2+2\tabcolsep},minimum height=\baselineskip+\extrarowheight] (box) {};
  \draw (box.north west) -- (box.south east);
  \draw (box.south west) -- (box.south west);
  \node[anchor=south west] at (box.south west) {#3};
  \node[anchor=north east] at (box.north east) {#4};
 \end{tikzpicture}}$\hskip-\tabcolsep}}


%% Use the graphics package to include figures
%\usepackage{graphics}

%% Use natbib with these recommended options
%\usepackage[authoryear,comma,longnamesfirst,sectionbib]{natbib}
\usepackage[authoryear,comma,sectionbib]{natbib} 

%%%%%%%%%%% Dannemiller added 01/19/2015 - standard preamble
\usepackage{fancyhdr} % Headers and footers

\pagestyle{fancyplain} % All pages have headers and footers
\fancyhead{} % Blank out the default header
\fancyfoot{} % Blank out the default footer
\fancyhead[RO,RE]{\thepage} % Custom header text
\renewcommand{\headrulewidth}{0pt} %0pt for no rule, 2pt thicker etc...
\setlength{\headheight}{15.0pt}

\usepackage{url}   %this allows us to cite URLs in the text

\usepackage{booktabs}
\usepackage{pstricks}
\let\clipbox\relax
\usepackage{amssymb}
\usepackage{afterpage}
\usepackage{transparent}
\usepackage{multirow}
\usepackage{rotating}
\usepackage{multicol}
%\usepackage{chngpage}


\usepackage{enumerate}
\usepackage{xspace}

\usepackage{abbrevs}
% example: \newabbrev\ART{American Repetrory Theater (ART)}[ART]
\newabbrev\ssb{Sum of Squares Between genotypes ($SS_{B}$)}[$SS_{B}$]
\newabbrev\ssw{Sum of Squares Within genotypes ($SS_{W}$)}[$SS_{W}$]
\newabbrev\sst{Sum of Squares Total ($SS_{T}$)}[$SS_{T}$]
\newabbrev\ci{current, intermediate (CI)}[CI]
\newabbrev\cim{current, intermediate matrix (CI matrix)}[CI matrix]
\newabbrev\cims{current, intermediate matrices (CI matrices)}[CI matrices]
\newabbrev\civ{current, intermediate vector (CI vector)}[CI vector]
\newabbrev\civs{current, intermediate vectors (CI vectors)}[CI vectors]
\newabbrev\cis{current, intermediate scalar (CI scalar)}[CI scalar]
\newabbrev\maf{minor allele frequency (MAF)}[MAF]



%%% The following correct for a flawed if statement in package abbrev
\makeatletter
\renewcommand\maybe@space@{%
  % \@tempswatrue % <= this is in the original
  \maybe@ictrue % <= this is new
  \expandafter   \@tfor
    \expandafter \reserved@a
    \expandafter :%
    \expandafter =%
                 \nospacelist
                 \do \t@st@ic
  % \if@tempswa % <= this is in the original
  \ifmaybe@ic % <= this is new
    \space
  \fi
}
\makeatother
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\usepackage[framemethod=TikZ]{mdframed}
\usepackage{ textcomp }
\newcommand\dblquote[1]{\textquotedblleft #1\textquotedblright} 
\newcommand{\apos}{$\text{'}$}
\newcommand{\be}{\begin{enumerate}}
\newcommand{\ee}{\end{enumerate}}
\newcommand{\bes}{\begin{enumerate*}}
\newcommand{\ees}{\end{enumerate*}}
\newcommand{\mb}{\noindent\makebox[\textwidth] }
\newcommand{\den}{\hspace{2pt}\textendash \,}
\newcommand{\dem}{\textemdash \,}
\newcommand{\Eq}{Equation }



%specific hyphenation
\hyphenation{e-pi-neph-rine}
\hyphenation{mol-e-cules}
%\hyphenation{}


\providecommand\phantomsection{}


%\usepackage{tikz}
\usepackage{xcolor}
\usepackage{hyperref}
\hypersetup{ colorlinks=false,pdfborderstyle={/S/U/W 1},citebordercolor=0 0 1 }
\colorlet{transpyellow}{white!10!yellow}

\usepackage{rotate}
\usepackage{rotating}
\usepackage{rotfloat}

\usepackage{fontspec}
\usepackage[lofdepth,lotdepth]{subfig}
\usepackage{threeparttable} 
\usepackage{tablefootnote}
\usepackage{footnote}

\usepackage{array}
%\newcolumntype{P}[1]{>{\raggedright\arraybackslash}p{#1}}
\newcolumntype{P}[1]{>{\centering\arraybackslash}p{#1}}
\newcolumntype{C}[1]{>{\centering\arraybackslash}p{#1}}

\usepackage{threeparttablex}


\usepackage{pdflscape} 
\usepackage{tabulary}
\usepackage{ltxtable}

\newcolumntype{Y}{>{\centering\arraybackslash}X}
\setlength\LTleft{0pt}
\setlength\LTright{0pt}



\usepackage{colortbl}
\definecolor{riceblue}{hsb}{0.6,0.1,0.9}
\definecolor{greenish}{rgb}{0.3,0.6,0.0}
\definecolor{myblue}{rgb}{0.3,0.3,0.8}
\definecolor{kugray5}{RGB}{224,224,224}
\definecolor{lightgray}{gray}{0.9}

\usepackage{xltxtra}
\usepackage{nicefrac}

\usepackage{rotating}
\usepackage{lscape}

\usepackage{graphicx}
%\usepackage{epstopdf}
\graphicspath{/Users/dannemil/paperless/ase_manuscript/sagmb/}

\usepackage[normal]{engord}

\usepackage{enumitem}
%\setlist[enumerate]{parsep=4pt}

\usepackage [normalem] {ulem}

\usepackage{tabularx}
\newcolumntype{S}{@{\stepcounter{Definition}\theDefinition.~} >{\bfseries}l @{~--~}X@{}}
\newcounter{Definition}[subsubsection]

\usepackage{longtable}
\usepackage{multirow}
\usepackage{caption}
%\renewcommand{\tablename}{Table}
\usepackage{threeparttable}

\usepackage{setspace}% http://ctan.org/pkg/setspace
\AtBeginEnvironment{tabular}{\singlespacing}% Single spacing in tabular environment

%\usepackage{booktabs}
\usepackage[detect-all]{siunitx}
\robustify\bfseries

\usepackage{tabularx,colortbl}
\newcolumntype{Y}{>{\centering\arraybackslash} X}
\renewcommand{\arraystretch}{1.5}

\usepackage{amsmath, amsthm, amssymb}
%\setlength{\mathindent}{0pt}

\usepackage{etoolbox}
\apptocmd\normalsize{%
 \abovedisplayskip=12pt plus 3pt minus 9pt
 \abovedisplayshortskip=0pt plus 3pt
 \belowdisplayskip=12pt plus 3pt minus 9pt
 \belowdisplayshortskip=7pt plus 3pt minus 4pt
}{}{}


%\numberwithin{figure}{section}

\makeatletter
\@addtoreset{footnote}{section}
\makeatother

\usepackage{mdwlist}


\renewcommand*\thesubsection{\arabic{section}.\arabic{subsection}}
\usepackage[title]{appendix}
\newcounter{appendix}
\numberwithin{equation}{appendix}
\usepackage{tablefootnote}
\usepackage{yhmath}
\usepackage{ragged2e}

\newcolumntype{C}[1]{>{\Centering\arraybackslash}p{#1}}

\usepackage{makecell}
\renewcommand\theadfont{\normalsize\bfseries}
\renewcommand\theadalign{bc}
\usepackage{cellspace}
\setlength\cellspacetoplimit{5pt}
\setlength\cellspacebottomlimit{5pt}

\setlength{\LTpre}{0pt}
\setlength{\LTpost}{12pt}
%\setlength\LTleft\parindent
%\setlength\LTright\fill


%\setmainfont{Times}
\setmainfont{Baskerville}
%\setmainfont{Alegreya}
%\setmainfont{Cardo}
\renewcommand{\vec}[1]{\mathbf{#1}}
\newcommand{\mtx}[1]{\mathbf{#1}}



\DeclareMathSizes{12pt}{11pt}{8pt}{6pt}
\everymath=\expandafter{\the\everymath\displaystyle}

\setcounter{secnumdepth}{5}

\usepackage{footnote}
\makesavenoteenv{tabular}
\makesavenoteenv{table}
\usepackage{tablefootnote}
\usepackage{adjustbox}
\setlength\heavyrulewidth{.06em}
\setlength\lightrulewidth{.04em}



\begin{document}

\noindent

Genes co-expressed with {\textit {COMT}} (n = 500) from http://coxpresdb.jp/ that overlap in each brain area with the 500 most strongly {\textit {COMT}}\den correlated genes (ranked by p-value) from the NIH data set.\\

{\small {
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:51:07 2018
\begin{table}[ht]
\centering
\begin{tabular}{rlr}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & ACAA1 &   3 \\ 
  2 & ACO2 &  22 \\ 
  3 & BCAN &   1 \\ 
  4 & BCAT2 &  19 \\ 
  5 & CAT &  11 \\ 
  6 & COMT &  22 \\ 
  7 & COPE &  19 \\ 
  8 & ECH1 &  19 \\ 
  9 & FAH &  15 \\ 
  10 & FUCA2 &   6 \\ 
  11 & GRHPR &   9 \\ 
  12 & MGST1 &  12 \\ 
  13 & MIF4GD &  17 \\ 
  14 & MSRB2 &  10 \\ 
  15 & PGLS &  19 \\ 
  16 & PPIB &  15 \\ 
  17 & RPS9 &  19 \\ 
  18 & SERF2 &  15 \\ 
  19 & SLC25A1 &  22 \\ 
  20 & TAX1BP3 &  17 \\ 
  21 & TECR &  19 \\ 
  22 & TST &  22 \\ 
  23 & TUFM &  16 \\ 
  24 & UQCR10 &  22 \\ 
  25 & YIF1A &  11 \\ 
  26 & ZMAT5 &  22 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 positively COMT - correlated genes in Prefrontal} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:51:07 2018
\begin{table}[ht]
\centering
\begin{tabular}{rlr}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & ACO2 &  22 \\ 
  2 & CAT &  11 \\ 
  3 & CNPY3 &   6 \\ 
  4 & CNTN2 &   1 \\ 
  5 & COMT &  22 \\ 
  6 & COPE &  19 \\ 
  7 & CRTAP &   3 \\ 
  8 & DESI1 &  22 \\ 
  9 & ECH1 &  19 \\ 
  10 & FUCA2 &   6 \\ 
  11 & GMPR2 &  14 \\ 
  12 & GRHPR &   9 \\ 
  13 & PEPD &  19 \\ 
  14 & PIGT &  20 \\ 
  15 & PPIB &  15 \\ 
  16 & RNH1 &  11 \\ 
  17 & SERF2 &  15 \\ 
  18 & TST &  22 \\ 
  19 & YIF1B &  19 \\ 
  20 & ZMAT5 &  22 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 positively COMT - correlated genes in Cerebellum} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:51:07 2018
\begin{table}[ht]
\centering
\begin{tabular}{rlr}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & ACAA1 &   3 \\ 
  2 & ACO2 &  22 \\ 
  3 & BCAN &   1 \\ 
  4 & CAT &  11 \\ 
  5 & CNPY3 &   6 \\ 
  6 & COMT &  22 \\ 
  7 & COPE &  19 \\ 
  8 & DNASE2 &  19 \\ 
  9 & ECH1 &  19 \\ 
  10 & FAH &  15 \\ 
  11 & GRHPR &   9 \\ 
  12 & GSTK1 &   7 \\ 
  13 & KIAA0930 &  22 \\ 
  14 & MGST1 &  12 \\ 
  15 & MIF4GD &  17 \\ 
  16 & MSRB2 &  10 \\ 
  17 & PCBD1 &  10 \\ 
  18 & PEPD &  19 \\ 
  19 & PPIB &  15 \\ 
  20 & SERF2 &  15 \\ 
  21 & TECR &  19 \\ 
  22 & TST &  22 \\ 
  23 & ZMAT5 &  22 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 positively COMT - correlated genes in Temporal} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:51:07 2018
\begin{table}[ht]
\centering
\begin{tabular}{rlr}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & CAT &  11 \\ 
  2 & CERS2 &   1 \\ 
  3 & CNPY3 &   6 \\ 
  4 & COMT &  22 \\ 
  5 & COPE &  19 \\ 
  6 & ELOVL1 &   1 \\ 
  7 & GRHPR &   9 \\ 
  8 & MIF4GD &  17 \\ 
  9 & PGLS &  19 \\ 
  10 & PIGT &  20 \\ 
  11 & PPIB &  15 \\ 
  12 & RNH1 &  11 \\ 
  13 & SERF2 &  15 \\ 
  14 & YIF1A &  11 \\ 
  15 & ZMAT5 &  22 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 positively COMT - correlated genes in Pons} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:54:45 2018
\begin{table}[ht]
\centering
\begin{tabular}{rlr}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & SMARCA2 &   9 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 negatively COMT - correlated genes in Prefrontal} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:54:45 2018
\begin{table}[ht]
\centering
\begin{tabular}{rlr}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & ARHGAP44 &  17 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 negatively COMT - correlated genes in Cerebellum} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:54:45 2018
\begin{table}[ht]
\centering
\begin{tabular}{rll}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 &  &  \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 negatively COMT - correlated genes in Temporal} 
\end{table}

\clearpage
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Thu May 31 08:54:45 2018
\begin{table}[ht]
\centering
\begin{tabular}{rll}
  \hline
 & geneSymbol & chromosome \\ 
  \hline
1 & ARHGAP44 & 17 \\ 
  2 & BCAP31 & X \\ 
  3 & KIAA0100 & 17 \\ 
  4 & PDXK & 21 \\ 
   \hline
\end{tabular}
\caption{Genes co-expressed with COMT from http://coxpresdb.jp/ matching the top 500 negatively COMT - correlated genes in Pons} 
\end{table}

}}
\end{document}