% Use this file as an \include to put it at the top of Sweave/R generated latex tables.

\documentclass[letterpaper,12pt]{article}

%\usepackage{dgjournal}


%%% Note: remove the next two lines to submit article %%%%%%%%%%
\usepackage{setspace}
%\linespread{2}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%       

%% The mathptmx package is recommended for Times compatible math symbols.
%% Use mtpro2 or mathtime instead of mathptmx if you have the commercially
%% available MathTime fonts.
%% Other options are txfonts (free) or belleek (free) or TM-Math (commercial)
\usepackage{mathptmx}
\usepackage{tensor}
\usepackage{nicefrac}
\usepackage{units}
\usepackage{mathtools}
\usepackage{mathrsfs}
\DeclareMathAlphabet{\mathpzc}{OT1}{pzc}{m}{it}
\usepackage{nth}
\usepackage{epsfig}
\usepackage[makeroom]{cancel}
\newcommand\Ccancel[2][black]{\renewcommand\CancelColor{\color{#1}}\cancel{#2}}
\newcommand\Bcancel[2][black]{\renewcommand\CancelColor{\color{#1}}\bcancel{#2}}

\usepackage{geometry}
\usepackage{changepage}



\usepackage[T1]{fontenc}
\usepackage{makecell}
\newcolumntype{x}[1]{>{\centering\arraybackslash}p{#1}}

\usepackage{tikz}
\newcommand\diag[4]{%
  \multicolumn{1}{p{#2}|}{\hskip-\tabcolsep
  $\vcenter{\begin{tikzpicture}[baseline=0,anchor=south west,inner sep=#1]
  \path[use as bounding box] (0,0) rectangle (#2+2\tabcolsep,\baselineskip);
  \node[minimum width={#2+2\tabcolsep},minimum height=\baselineskip+\extrarowheight] (box) {};
  \draw (box.north west) -- (box.south east);
  \draw (box.south west) -- (box.south west);
  \node[anchor=south west] at (box.south west) {#3};
  \node[anchor=north east] at (box.north east) {#4};
 \end{tikzpicture}}$\hskip-\tabcolsep}}


%% Use the graphics package to include figures
%\usepackage{graphics}

%% Use natbib with these recommended options
%\usepackage[authoryear,comma,longnamesfirst,sectionbib]{natbib}
\usepackage[authoryear,comma,sectionbib]{natbib} 

%%%%%%%%%%% Dannemiller added 01/19/2015 - standard preamble
\usepackage{fancyhdr} % Headers and footers

\pagestyle{fancyplain} % All pages have headers and footers
\fancyhead{} % Blank out the default header
\fancyfoot{} % Blank out the default footer
\fancyhead[RO,RE]{\thepage} % Custom header text
\renewcommand{\headrulewidth}{0pt} %0pt for no rule, 2pt thicker etc...
\setlength{\headheight}{15.0pt}

\usepackage{url}   %this allows us to cite URLs in the text

\usepackage{booktabs}
\usepackage{pstricks}
\let\clipbox\relax
\usepackage{amssymb}
\usepackage{afterpage}
\usepackage{transparent}
\usepackage{multirow}
\usepackage{rotating}
\usepackage{multicol}
%\usepackage{chngpage}


\usepackage{enumerate}
\usepackage{xspace}

\usepackage{abbrevs}
% example: \newabbrev\ART{American Repetrory Theater (ART)}[ART]
\newabbrev\ssb{Sum of Squares Between genotypes ($SS_{B}$)}[$SS_{B}$]
\newabbrev\ssw{Sum of Squares Within genotypes ($SS_{W}$)}[$SS_{W}$]
\newabbrev\sst{Sum of Squares Total ($SS_{T}$)}[$SS_{T}$]
\newabbrev\ci{current, intermediate (CI)}[CI]
\newabbrev\cim{current, intermediate matrix (CI matrix)}[CI matrix]
\newabbrev\cims{current, intermediate matrices (CI matrices)}[CI matrices]
\newabbrev\civ{current, intermediate vector (CI vector)}[CI vector]
\newabbrev\civs{current, intermediate vectors (CI vectors)}[CI vectors]
\newabbrev\cis{current, intermediate scalar (CI scalar)}[CI scalar]
\newabbrev\maf{minor allele frequency (MAF)}[MAF]



%%% The following correct for a flawed if statement in package abbrev
\makeatletter
\renewcommand\maybe@space@{%
  % \@tempswatrue % <= this is in the original
  \maybe@ictrue % <= this is new
  \expandafter   \@tfor
    \expandafter \reserved@a
    \expandafter :%
    \expandafter =%
                 \nospacelist
                 \do \t@st@ic
  % \if@tempswa % <= this is in the original
  \ifmaybe@ic % <= this is new
    \space
  \fi
}
\makeatother
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\usepackage[framemethod=TikZ]{mdframed}
\usepackage{ textcomp }
\newcommand\dblquote[1]{\textquotedblleft #1\textquotedblright} 
\newcommand{\apos}{$\text{'}$}
\newcommand{\be}{\begin{enumerate}}
\newcommand{\ee}{\end{enumerate}}
\newcommand{\bes}{\begin{enumerate*}}
\newcommand{\ees}{\end{enumerate*}}
\newcommand{\mb}{\noindent\makebox[\textwidth] }
\newcommand{\den}{\hspace{2pt}\textendash \,}
\newcommand{\dem}{\textemdash \,}
\newcommand{\Eq}{Equation }



%specific hyphenation
\hyphenation{e-pi-neph-rine}
\hyphenation{mol-e-cules}
%\hyphenation{}


\providecommand\phantomsection{}


%\usepackage{tikz}
\usepackage{xcolor}
\usepackage{hyperref}
\hypersetup{ colorlinks=false,pdfborderstyle={/S/U/W 1},citebordercolor=0 0 1 }
\colorlet{transpyellow}{white!10!yellow}

\usepackage{rotate}
\usepackage{rotating}
\usepackage{rotfloat}

\usepackage{fontspec}
\usepackage[lofdepth,lotdepth]{subfig}
\usepackage{threeparttable} 
\usepackage{tablefootnote}
\usepackage{footnote}

\usepackage{array}
%\newcolumntype{P}[1]{>{\raggedright\arraybackslash}p{#1}}
\newcolumntype{P}[1]{>{\centering\arraybackslash}p{#1}}
\newcolumntype{C}[1]{>{\centering\arraybackslash}p{#1}}

\usepackage{threeparttablex}


\usepackage{pdflscape} 
\usepackage{tabulary}
\usepackage{ltxtable}

\newcolumntype{Y}{>{\centering\arraybackslash}X}
\setlength\LTleft{0pt}
\setlength\LTright{0pt}



\usepackage{colortbl}
\definecolor{riceblue}{hsb}{0.6,0.1,0.9}
\definecolor{greenish}{rgb}{0.3,0.6,0.0}
\definecolor{myblue}{rgb}{0.3,0.3,0.8}
\definecolor{kugray5}{RGB}{224,224,224}
\definecolor{lightgray}{gray}{0.9}

\usepackage{xltxtra}
\usepackage{nicefrac}

\usepackage{rotating}
\usepackage{lscape}

\usepackage{graphicx}
%\usepackage{epstopdf}
\graphicspath{/Users/dannemil/paperless/ase_manuscript/sagmb/}

\usepackage[normal]{engord}

\usepackage{enumitem}
%\setlist[enumerate]{parsep=4pt}

\usepackage [normalem] {ulem}

\usepackage{tabularx}
\newcolumntype{S}{@{\stepcounter{Definition}\theDefinition.~} >{\bfseries}l @{~--~}X@{}}
\newcounter{Definition}[subsubsection]

\usepackage{longtable}
\usepackage{multirow}
\usepackage{caption}
%\renewcommand{\tablename}{Table}
\usepackage{threeparttable}

\usepackage{setspace}% http://ctan.org/pkg/setspace
\AtBeginEnvironment{tabular}{\singlespacing}% Single spacing in tabular environment

%\usepackage{booktabs}
\usepackage[detect-all]{siunitx}
\robustify\bfseries

\usepackage{tabularx,colortbl}
\newcolumntype{Y}{>{\centering\arraybackslash} X}
\renewcommand{\arraystretch}{1.5}

\usepackage{amsmath, amsthm, amssymb}
%\setlength{\mathindent}{0pt}

\usepackage{etoolbox}
\apptocmd\normalsize{%
 \abovedisplayskip=12pt plus 3pt minus 9pt
 \abovedisplayshortskip=0pt plus 3pt
 \belowdisplayskip=12pt plus 3pt minus 9pt
 \belowdisplayshortskip=7pt plus 3pt minus 4pt
}{}{}


%\numberwithin{figure}{section}

\makeatletter
\@addtoreset{footnote}{section}
\makeatother

\usepackage{mdwlist}


\renewcommand*\thesubsection{\arabic{section}.\arabic{subsection}}
\usepackage[title]{appendix}
\newcounter{appendix}
\numberwithin{equation}{appendix}
\usepackage{tablefootnote}
\usepackage{yhmath}
\usepackage{ragged2e}

\newcolumntype{C}[1]{>{\Centering\arraybackslash}p{#1}}

\usepackage{makecell}
\renewcommand\theadfont{\normalsize\bfseries}
\renewcommand\theadalign{bc}
\usepackage{cellspace}
\setlength\cellspacetoplimit{5pt}
\setlength\cellspacebottomlimit{5pt}

\setlength{\LTpre}{0pt}
\setlength{\LTpost}{12pt}
%\setlength\LTleft\parindent
%\setlength\LTright\fill


%\setmainfont{Times}
\setmainfont{Baskerville}
%\setmainfont{Alegreya}
%\setmainfont{Cardo}
\renewcommand{\vec}[1]{\mathbf{#1}}
\newcommand{\mtx}[1]{\mathbf{#1}}



\DeclareMathSizes{12pt}{11pt}{8pt}{6pt}
\everymath=\expandafter{\the\everymath\displaystyle}

\setcounter{secnumdepth}{5}

\usepackage{footnote}
\makesavenoteenv{tabular}
\makesavenoteenv{table}
\usepackage{tablefootnote}
\usepackage{adjustbox}
\setlength\heavyrulewidth{.06em}
\setlength\lightrulewidth{.04em}



\begin{document}

\noindent

\begin{landscape}

{\footnotesize {
\setlength\LTleft{-1cm}
\setlength\LTright{-1cm}
% latex table generated in R 3.4.4 by xtable 1.8-2 package
% Wed May 30 14:23:45 2018
\begin{longtable}{lrrrrrl}
\caption{CC Ontology for the positively correlated genes with the lowest p-values for expression correlations with COMT: Cerebellum} \\ 
  \toprule
GOCCID & Pvalue & OddsRatio & ExpCount & Count & Size & Term \\ 
  \midrule
GO:0044444 & 0.00000001239059108 & 1.86 & 193.93 & 246 & 8105 & cytoplasmic part \\ 
  GO:0031982 & 0.00000001728684747 & 1.88 & 88.84 & 136 & 3713 & vesicle \\ 
  GO:0005737 & 0.00000006386582233 & 1.90 & 232.43 & 279 & 9714 & cytoplasm \\ 
  GO:0070062 & 0.00000014553588190 & 1.94 & 57.76 & 96 & 2414 & extracellular exosome \\ 
  GO:1903561 & 0.00000019585713572 & 1.92 & 58.12 & 96 & 2429 & extracellular vesicle \\ 
  GO:0043230 & 0.00000020370408031 & 1.92 & 58.17 & 96 & 2431 & extracellular organelle \\ 
  GO:0012505 & 0.00000030790726679 & 1.78 & 89.42 & 132 & 3737 & endomembrane system \\ 
  GO:0005739 & 0.00000501294410292 & 1.98 & 34.86 & 62 & 1457 & mitochondrion \\ 
  GO:0044446 & 0.00000942084571567 & 1.60 & 184.51 & 225 & 7711 & intracellular organelle part \\ 
  GO:0005777 & 0.00002160577684402 & 4.83 & 2.78 & 12 & 116 & peroxisome \\ 
  GO:0042579 & 0.00002160577684402 & 4.83 & 2.78 & 12 & 116 & microbody \\ 
  GO:0031410 & 0.00002472512090279 & 1.78 & 46.35 & 74 & 1937 & cytoplasmic vesicle \\ 
  GO:0097708 & 0.00002560851123653 & 1.78 & 46.40 & 74 & 1939 & intracellular vesicle \\ 
  GO:0044421 & 0.00002980752678786 & 1.62 & 79.99 & 113 & 3343 & extracellular region part \\ 
  GO:0044438 & 0.00003110219612279 & 5.64 & 2.01 & 10 & 84 & microbody part \\ 
  GO:0044439 & 0.00003110219612279 & 5.64 & 2.01 & 10 & 84 & peroxisomal part \\ 
  GO:0005783 & 0.00003473474084198 & 1.84 & 37.11 & 62 & 1551 & endoplasmic reticulum \\ 
  GO:0044422 & 0.00004067050253296 & 1.54 & 188.72 & 226 & 7887 & organelle part \\ 
  GO:0043227 & 0.00004268016280975 & 1.65 & 252.63 & 286 & 10558 & membrane-bounded organelle \\ 
  GO:0044433 & 0.00004392590974809 & 1.91 & 30.34 & 53 & 1268 & cytoplasmic vesicle part \\ 
  GO:0044432 & 0.00005016082148455 & 1.95 & 27.42 & 49 & 1146 & endoplasmic reticulum part \\ 
  GO:0042765 & 0.00005338674445877 & 123.39 & 0.10 & 3 & 4 & GPI-anchor transamidase complex \\ 
  GO:0005615 & 0.00005625375914722 & 1.60 & 75.80 & 107 & 3168 & extracellular space \\ 
  GO:0005789 & 0.00006217802294359 & 2.06 & 21.65 & 41 & 905 & endoplasmic reticulum membrane \\ 
  GO:0098827 & 0.00006856137520693 & 2.05 & 21.75 & 41 & 909 & endoplasmic reticulum subcompartment \\ 
  GO:0042175 & 0.00009589113074606 & 2.01 & 22.09 & 41 & 923 & nuclear outer membrane-endoplasmic reticulum membrane network \\ 
   \bottomrule
\end{longtable}
}}
\end{landscape}
\end{document}